_fr/tour/singleton-objects.md

```diff
@@ -3,10 +3,118 @@ layout: tour
 title: Singleton Objects
 partof: scala-tour
 
-num: 12
+num: 15
 
 language: fr
 
 next-page: regular-expression-patterns
 previous-page: pattern-matching
 ---
+
+Un objet est une classe qui a exactement une instance. Il est créé de façon passive au moment où il est référencé, comme une valeur passive `lazy val`.
+
+En tant que valeur de premier niveau, un object est un singleton.
+
+En tant que membre d'une classe englobante ou en tant que valeur locale, il se comporte exactement comme une `lazy val`.
+
+# Définir un objet singleton
+
+Un objet est une valeur. La définition d'un objet ressemble a une classe, mais utilise le mot clef `object` :
+
+```scala mdoc
+object Box
+```
+
+Voici l'exemple d'un objet avec une méthode :
+
+```
+package logging
+
+object Logger {
+  def info(message: String): Unit = println(s"INFO: $message")
+}
+```
+
+La méthode `info` peut être importée depuis n'importe où dans le programme. Créer des méthodes utilitaires, comme celle-ci, est un cas d'usage commun pour les objets singleton.
+
+Regardons comment utiliser `info` dans un autre package :
+
+```
+import logging.Logger.info
+
+class Project(name: String, daysToComplete: Int)
+
+class Test {
+  val project1 = new Project("TPS Reports", 1)
+  val project2 = new Project("Website redesign", 5)
+  info("Created projects")  // Prints "INFO: Created projects"
+}
+```
+
+La méthode `info` est visible grâce à l'import, `import logging.Logger.info`. Les imports ont besoin d'un chemin d'accès stable aux ressources, et un objet est un chemin stable.
+
+Note : Si un `objet` est encapsulé dans une autre classe ou un autre objet, alors l'objet est dépendant du chemin d'accès comme les autres membres. Cela veut dire par exemple, que si on prend 2 types de boissons, `class Milk` et `class OrangeJuice`, un membre de class `object NutritionInfo` est dépendant de son instance d'encapsulation. `milk.NutritionInfo` est complétement différent de `oj.NutritionInfo`.
+
+## Les objets compagnons
+
+Un objet avec le même nom qu'une classe est appelé un _objet compagnon_. Inversement, la classe se nomme la _classe compagnon_ de l'objet. Une classe ou un objet compagnon peut accéder aux membres privés de son compagnon. 
```

Comment:
This sentence from the original does not appear to have been translated:

> Use a companion object for methods and values which are not specific to instances of the companion class.

Reply:
I knew that I missed a sentence, but I was thinking that I found it. Ty, I fixed it in the last commit with :

> L'objet compagnon est utile pour les méthodes et les valeurs qui ne sont pas spécifiques aux instances de la classe compagnon.